src_c/rect.c

```diff
@@ -307,6 +297,7 @@ pgRect_FromObject(PyObject *obj, SDL_Rect *temp)
             PyObject *rectresult = PyObject_CallObject(rectattr, NULL);
```

Comment:
a few lines above here, `PyObject_GetAttrString` returns `rectattr`, and this can be `NULL` on error with python error set. We need to handle that case by clearing the error and returning `NULL` before doing a `PyCallable_Check` here